lib/instrumenter/README.md

```diff
@@ -1,3 +1,65 @@
+
+## API
+
+The primary way to use the Storybook Instrumenter is through the `instrument` function:
+
+```ts
+instrument<TObj extends Record<string, any>>(obj: TObj, options: Options): TObj
+```
+
+`instrument` takes a plain JS object or imported ES module, and optionally an `options` object. It traverses the input object, recursively iterating over object properties and arrays. Any values with typeof `function` are tracked (through monkey-patching). Finally, a shallow copy of the original object is returned (with functions replaced). If the `mutate: true` option is set, the original object is mutated instead of returning a shallow copy.
+
+### Options
+
+- `intercept: boolean | ((method: string, path: Array<string | CallRef>) => boolean)` - Whether to make functions interceptable
+- `retain: boolean` - Whether to retain calls across renders (e.g. for story setup functions / loaders that run only once)
+- `mutate: boolean` - Whether to mutate the input object instead of returning a shallow copy
+- `path: Array<string | CallRef>` - A virtual object path to prepend to the actual input object function paths
+- `getArgs: (call: Call, state: State) => Call['args']` - Allows overriding args before invoking the original function with them
+
+## Events
+
+The Storybook Instrumenter uses the [Storybook Channel API](../channels/README.md) to send and receive events.
+
+### Emitted tracking events
+
+The instrumenter emits two types of events for tracking function invocations ("calls"):
+
+- `storybook/instrumenter/call` - Emitted whenever a tracked function is invoked
+- `storybook/instrumenter/sync` - Emitted after one or more tracked functions are invoked (batch-wise)
+
+The `storybook/instrumenter/call` event payload contains all metadata about the function invocation, including a unique `id`, any arguments, the method name and object path. However, the order of events is not guaranteed and you may receive the same call multiple times while debugging. Moreover, this event is emitted for _all_ tracked calls, not just interceptable ones.
+
+The `storybook/instrumenter/sync` event payload contains a list of `logItems` which represents a "normalized" log of _interceptable_ calls. The order of calls is guaranteed and step-through debugging will not append to the log but rather update it to set the proper `status` for each call. The log does not contain full call metadata but only a `callId`, so this must be mapped onto received `storybook/instrumenter/call` events. The `storybook/instrumenter/sync` event also contains `callStates`, see below.
```

Comment:
This could use a detailed example if we want people to actually use this package to instrument things